### ISSUE TITLE
Clone an archive within the repo

### ISSUE THREAD

**Issue by henrus** [borgbackup/borg#7254]
Dear All,

thank you for the excellent work.

I am using borg for years and the repo is growing. A lot of the data in my backups are computational results which I would like to strip out of the repo after some time. This can be done using borg recreate. So far so good ...

What I am missing is a method to clone an archive. The content of the archive would be identical just under a different name. Cloning would give me two options: a) It easier and more flexible to specify the retention policies on the two sets of archives; b) I could test the recreation without the danger of destroying anything.

The invocation should be something like

borg clone repo1::oldArchive newArchive

## Have you checked borgbackup docs, FAQ, and open Github issues?

There have been many requests to copy an archive from one repo to another which culminated in borg transfer. However, this is not what I am asking for.

## Is this a BUG / ISSUE report or a QUESTION?

QUESTION / SUGGESTION

**Comment by infectormp** [comment]
You need to use `borg transfer', but this command is only available in version 2.0, which is still a beta version. 
https://borgbackup.readthedocs.io/en/2.0.0b4/usage/transfer.html

**Comment by henrus** [comment]
Dear infectormp,

thank you for your quick reply. I am aware of borg transfer (see 
original post). However, borg transfer is NOT quite want I want because 
it transfers archives from one repository to another repository (see 
documentation) where I would like to clone (transfer) within the same repo.

Best Regards,
Henrik

**Comment by ThomasWaldmann** [comment]
https://borgbackup.readthedocs.io/en/stable/usage/recreate.html

If you use the `--target <newname>` option (and otherwise you don't use any recreate option that actually changes anything in the archive while recreating it), the result will be a clone of the original archive and it won't delete the original archive after recreating it.

In general, be very careful with recreate so you don't accidentally lose data (e.g. by accidentally excluding it).

**Comment by ThomasWaldmann** [comment]
BTW, an alternative approach to stripping not needed-any-more data from existing archives could be to not put that data into a single archive with the stuff you want to keep. Then you could just delete an archive instead of having to use recreate.

As an example, I use that approach to backup /home (user data) separately from / (system data), creating 2 sequences of archives:
- hostname-home-timestamp
- hostname-system-timestamp

I then use 2 different prune invocations (important is to use the corresponding `--prefix` or `--glob-archives` to only select the archives of 1 sequence) to quickly prune the system stuff and keep the home stuff much longer.

**Comment by henrus** [comment]
Hi,
indeed, borg recreate --target is what I need. Sorry, for not spotting this and wasting your time.

I considered two streams of archives. The problem is that the patterns are rather unusual as the code uses the file system as a database. Firstly, I need to locate the directories where computational data is stored - two directories need to exist in this directory; let's call it CASE. I then need to exclude directories relative to CASE those names match floating point numbers, but not the first one in numerical order - so "-0.1" shall be included in preference to "-0.01". And this is not the end of it ...

The complex pattern can be dealt with in two ways: a) I can generate the exclude pattern before starting the backup. Here, the problem is that new data might be generated during the backup, so it will include some extra files; b) Some dynamic extension to the pattern matcher.

Anyway, the (complex) pattern matching can go wrong in various ways and I did not want to learn the hard way that some important data has been filtered out over years ;-) That's why I settled on applying a minimal exclude pattern (only some caches) for the initial backup. However, now I need to find a way to reduce disk space/chunk cache.

The idea is to create the exclude pattern from the archive, but maybe there is a better way that saves me a pass through the files ...

Best Regards,
Henrik

**Comment by ThomasWaldmann** [comment]
If it is easier to generate a list of files outside of borg (by some tool or own script), that can be fed into borg using `--paths-from-stdin`, disabling the borg internal recursion and pattern matching.

Yeah, it is usually better to back up more than needed than accidentally miss something.

**Comment by bof** [comment]
I've been testing just that (recreate --transfer) with borg2 b4, a few days ago, and found it only actually did something when I added a --comment option to the call.

Wasn't sure whether that's intentional, so didn't make a bug report. Should I?

**Comment by ThomasWaldmann** [comment]
@bof that's a bug, thanks for reporting it.

the condition detecting "nothing to do" did not consider the `--target` option. I'll fix it.

**Comment by ThomasWaldmann** [comment]
@henrus 1.2-maint has the same issue as @bof found for 2.0 beta. I'll fix it.

Until a fix for 1.2.x is released, one needs to give e.g. `--comment "some comment"` to work around that issue.

**Comment by henrus** [comment]
Hi,
I tried
borg recreate REPO::old --exclude NIX --target new
with borg version 1.2.3
It does not seem to update the chunk cache correctly
Archives: 17, w/ cached Idx: 16, w/ outdated Idx: 0, w/o cached Idx: 1.
And then has to recreate it.
Is this a know problem or shall I write a proper issue?
Henrik

**Comment by ThomasWaldmann** [comment]
@henrus maybe rather open a separate issue about that.

**Comment by ThomasWaldmann** [comment]
The bug discovered here is fixed, the question is answered, so guess nothing else to do here.
